Database/Internal.hs

```diff
@@ -23,6 +27,13 @@ data Database = Database { dbConn :: Connection
                          , dbRNG :: MVar HashDRBG
                          }
 
+type DatabaseM a = ReaderT Database IO a
```

Comment:
Think of this as the following typealias:

`type DatabaseM a = Database -> IO a`

That's really all it is, but the `ReaderT` thing is in there so that `>>=` can be overloaded to Do The Right Thing. Unfortunately this hides the simplicity of `Reader` -- it's literally just a function -- and makes it a little scarier looking. But don't be afraid!